ConsoleApp/SnakeGame.cs

```diff
@@ -0,0 +1,616 @@
+                {
+                    _stack.Push(new Snack(_head._position.x, _head._position.y, _food[_head._position.y].value));
+                    _foodLength++;
+                    _food[_head._position.y] = null;
+                    return;
+                }
+            }
+
+            _stack.CopyTo(_stackCopy);
+            _head = new Snack(_stackCopy[_stackCopy.Length - 1]._position.x, _stackCopy[_stackCopy.Length - 1]._position.y, _stackCopy[_stackCopy.Length - 1].value);
+
+            try
+            {
+                if (_moveTo(_head, ref _stackCopy) != true)
+                {
+                    ShowDebugData(_stackCopy, _head, "Snake is stacked!");
+                    return;
+                }
+
+                Console.SetCursorPosition(_stackCopy[0]._position.x, _stackCopy[0]._position.y);
+                Console.Write(' ');
+                Console.SetCursorPosition(_head._position.x, _head._position.y);
+                Console.Write(_head.value);
+
+                _stack.Clear();
+                for (i = 0; i < _stackCopy.Length - 1; i++)
+                {
+                    _stack.Push(new Snack(_stackCopy[i + 1]._position.x, _stackCopy[i + 1]._position.y, _stackCopy[i].value));
+                    Console.SetCursorPosition(_stackCopy[i + 1]._position.x, _stackCopy[i + 1]._position.y);
+                    Console.Write(_stackCopy[i].value);
+                }
+                _stack.Push(new Snack(_head._position.x, _head._position.y, _head.value));
+                ShowDebugData(_stackCopy, _head, "                  ");
+            }
+            catch (Exception ex)
+            {
+                ShowDebugData(_stackCopy, _head, "                  ");
+                Console.WriteLine(ex.ToString());
+            }
+        }
+        private void ShowDebugData(Snack[] _stackCopy, Snack _head, string? error)
+        {
+            Console.SetCursorPosition(0, _axeY + 1);
+            Console.WriteLine("{0} x {1} : {2} x {3}  Mode: {4}   {5}   ", _point.X.Length, _point.Y.Length, _head._position.x, _head._position.y, _modeMove, error);
+            Console.Write("Snack Data: \"");
+            for (int i = _stackCopy.Length - 1; i >= 0; i--)
+            {
+                Console.Write(_stackCopy[i].value);
+            }
+            Console.Write("\", Length: {0}    \r\n\r\n", _stackCopy.Length);
+            // Console.WriteLine("Error: {0}    \r\n", error);
+        }
+        public void SetNewDirection(object obj)
+        {
+            Random r = new Random();
+            Move rMove = (Move)r.Next(0, 4);
+            _setModeMove = rMove;
+        }
+        private bool _moveTo(Snack _head, ref Snack[] stackCopy)
+        {
+            bool _enableMove = false;
+
+            switch (_setModeMove)
+            {
+                case Move.UP:
+                    {
+                        _modeMove = Move.UP;
```

Comment:
almost all code in case below copypasted, I think it's possible to make it simplier